cul-it/arxiv-compiler
# ARXIVNG-1324 ARXIVNG-1325 minimum functional integration with flask, celery

This delivers:
- ARXIVNG-1324 stand up celery worker to test async compilation
- ARXIVNG-1325 stand up web app to handle api requests and create tasks for celery

I couldn't find any commits related to ARXIVNG-1323, so I created a dummy task and creation/status helper functions in ``compiler.compiler``. Just need to swap in the real task name in ``compiler.compiler.create_compilation_task``.

To test manually:
- Install the latest version of docker-compose: ``pip install -U docker-compose``
- Build and run with docker-compose:
```bash
docker-compose build
docker-compose up
```
- You can make requests to the ``requestCompilation`` operation:

```bash
$ curl -XPOST -i -d '{"source_id":"123","checksum":"a1b2c3etc","format":"pdf"}' http://localhost:8000/
HTTP/1.1 202 ACCEPTED
Content-Type: application/json
Content-Length: 3
Location: http://localhost:8000/task/97f0df86-4d52-491d-9607-a896f590249b

{}
```

- And check the status of the compilation task (but note that we're returning HTML, so that should get fixed):

```bash
$ curl -XGET -i http://localhost:8000/task/97f0df86-4d52-491d-9607-a896f590249b
HTTP/1.1 303 SEE OTHER
Content-Type: text/html; charset=utf-8
Content-Length: 243
Location: http://localhost:8000/123/a1b2c3etc/pdf

<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 3.2 Final//EN">
<title>Redirecting...</title>
<h1>Redirecting...</h1>
<p>You should be redirected automatically to target URL: <a href="/123/a1b2c3etc/pdf">/123/a1b2c3etc/pdf</a>.  If not click the link.
```
